### PR TITLE
商品一覧画面のソート順選択時に表示が変更されるように修正しました

### PR DESCRIPTION
**Pull request by SekiMoemi**
以下のように変更しました。ご確認お願いします。
・ソート条件の選択時にページが更新されるように変更しました
・検索条件をフォームから受け取るように変更しました
・検索後も条件を表示されるようにHTMLを修正しました
・クリアボタンを押したときに検索のおもちゃ名をリセットするように修正しました

### CODE REVIEW

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 
 import jp.co.example.ecommerce_a.domain.Item;
+import jp.co.example.ecommerce_a.form.SearchItemForm;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:13]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'jp.co.example.ecommerce_a.form.SearchItemForm' import. Should be before 'org.springframework.web.bind.annotation.RequestMapping'.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 2]
```diff
@@ -28,16 +29,16 @@ public class ShowItemListController {
 	/**
 	 * おもちゃ一覧画面を出力します.
 	 * 
-	 * @param name  検索文字列
+	 * @param form  商品の検索時に利用するフォーム
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:32]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:32]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 3]
```diff
@@ -28,16 +29,16 @@ public class ShowItemListController {
...
 	 * @param model モデル
 	 * @return おもちゃ一覧画面
 	 */
 	@GetMapping("")
-	public String showItemList(String name, Model model, String sortNum) {
-		List<Item> itemList = showItemListService.searchByName(name, sortNum);
+	public String showItemList(SearchItemForm form, Model model) {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:37]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 4]
```diff
@@ -28,16 +29,16 @@ public class ShowItemListController {
...
+		List<Item> itemList = showItemListService.searchByName(form.getName(), form.getSortNum());
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 106).

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 5]
```diff
@@ -28,16 +29,16 @@ public class ShowItemListController {
...
 		// 見つけない場合、全件検索する
 		if (itemList.size() == 0) {
-			itemList = showItemListService.searchByName("", sortNum);
+			itemList = showItemListService.searchByName("", form.getSortNum());
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
+package jp.co.example.ecommerce_a.form;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:1]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.PackageNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Package name 'jp.co.example.ecommerce_a.form' must match pattern '^[a-z]+(\.[a-z][a-z0-9]*)*$'.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * 商品の検索時に利用するフォーム.
+ * 
+ * @author moemi.seki
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:6]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@author' should be preceded with an empty line.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ */
+public class SearchItemForm {
+	/** 商品名 */
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:10]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:10]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+	private String name;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:11]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:11]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'member def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+	/** ソート番号 */
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:12]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:12]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+	private String sortNum;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:13]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:13]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'member def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+	public String getName() {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:15]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:15]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+		return name;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:16]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:16]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+	}
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:17]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:17]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def rcurly' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+	public void setName(String name) {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:19]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:19]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+		this.name = name;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:20]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:20]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+	}
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java:21]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

**Review by SekiMoemi — COMMENTED** [review]
Remaining comments which cannot be posted as a review comment to avoid GitHub Rate Limit

<details>
<summary>checkstyle</summary>

[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|21 col 9|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L21) 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|23 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L23) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|23 col 9|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L23) 'method def modifier' has incorrect indentation level 8, expected level should be 2.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|24 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L24) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|24 col 17|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L24) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|25 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L25) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|25 col 9|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L25) 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|27 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L27) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|27 col 9|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L27) 'method def modifier' has incorrect indentation level 8, expected level should be 2.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|28 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L28) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|28 col 17|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L28) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|29 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L29) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|29 col 9|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L29) 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|31 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L31) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|31 col 9|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L31) 'method def modifier' has incorrect indentation level 8, expected level should be 2.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|32 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L32) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|33 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L33) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|33 col 17|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L33) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|34 col 1|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L34) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java|34 col 9|](http://github.com/ecsite-project/ec-202304a/blob/575ce0ba7f9b02e80403feffdbbb9131b262db19/src/main/java/jp/co/example/ecommerce_a/form/SearchItemForm.java#L34) 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
</details>

**Review by yamada-kensyu — APPROVED** [review]
承認します！

**Pull request merged**